File: src/visual-graph/prune/push-to.ts
export function pushTo(
  map: Map<string, /* mutable */ string[]>,
  key: string,
  value: string,
): void {
  const arr = map.get(key) ?? null;
  if (arr === null) {
    map.set(key, [value]);
  } else {
    arr.push(value);
  }
}

File: crates/unsnarl-visual-graph/src/prune/push_to.rs
//! Append-or-create helper for `Map<String, Vec<String>>`.
//!
//! Mirrors `ts/src/visual-graph/prune/push-to.ts`.

use std::collections::HashMap;

pub fn push_to(map: &mut HashMap<String, Vec<String>>, key: &str, value: String) {
    match map.get_mut(key) {
        Some(arr) => arr.push(value),
        None => {
            map.insert(key.to_string(), vec![value]);
        }
    }
}

#[cfg(test)]
#[path = "push_to_test.rs"]
mod push_to_test;
